File: leet/0101-symmetric-tree/symmetric-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    if root == nil {return true}
    if root.Left == nil && root.Right == nil {return true}
    
    list := make([]*TreeNode, 2)
    list[0] = root.Left
    list[1] = root.Right
    
    for len(list) != 0 {
        i := 0
        j := len(list) - 1
        for i < j {
            if((list[i] != nil && list[j] == nil) || list[i] == nil && list[j] != nil || (list[i] != nil && list[j] != nil && list[i].Val != list[j].Val) ){
                return false
            }
            
            i++; j--
        }
        
        temp := make([]*TreeNode, 0)
        
        for k := 0; k < len(list); k++ {
            if(list[k] != nil){
                temp = append(temp, list[k].Left)
                temp = append(temp, list[k].Right)
            }
        }
        
        list = temp
    }
    
    return true
}

File: leet/0101-symmetric-tree/symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    if (!root) return true
    
    let checkSymm = function(n1, n2){
        if (!n1 && !n2) return true
        if (!n1 || !n2) return false
        
        return n1.val == n2.val && checkSymm(n1.left, n2.right) && checkSymm(n1.right, n2.left)
    }
    
    return checkSymm(root.left, root.right)
};
